LevelComponents/Room.cpp

```diff
@@ -352,17 +352,25 @@ namespace LevelComponents
                         {
                             for (int k = 0; k < qMin(sceneWidth, imageA.width()); ++k)
                             {
-                                QColor PXA = QColor(imageA.pixel(k, j)), PXB = QColor(imageB.pixel(k, j));
-                                int R = qMin((Layer0ColorBlendCoefficient_EVA * PXA.red()) / 16 +
-                                                 (Layer0ColorBlendCoefficient_EVB * PXB.red()) / 16,
-                                             255);
-                                int G = qMin((Layer0ColorBlendCoefficient_EVA * PXA.green()) / 16 +
-                                                 (Layer0ColorBlendCoefficient_EVB * PXB.green()) / 16,
-                                             255);
-                                int B = qMin((Layer0ColorBlendCoefficient_EVA * PXA.blue()) / 16 +
-                                                 (Layer0ColorBlendCoefficient_EVB * PXB.blue()) / 16,
-                                             255);
-                                imageA.setPixel(k, j, QColor(R, G, B).rgb());
+                                if(imageA.pixelColor(k, j).alpha() == 0xFF && imageB.pixelColor(k, j).alpha() == 0xFF) // current pixels not transparent
```

Comment:
Let's extract this to a function (and the 2 other instances of the same logic)

Reply:
Done.